Firestore/core/src/local/leveldb_key.h

```diff
@@ -635,6 +643,76 @@ class LevelDbRemoteDocumentReadTimeKey {
   model::SnapshotVersion read_time_;
 };
 
+/**
+ * A key in the bundles table, storing the bundle Id for each entry.
+ */
+class LevelDbBundlesKey {
+ public:
+  /**
+   * Creates a key prefix that points just before the first key of the table.
+   */
+  static std::string KeyPrefix();
+
+  /**
+   * Creates a key that points to the key for the given bundle id.
+   */
+  static std::string Key(absl::string_view bundle_id);
+
+  /**
+   * Decodes the given complete key, storing the decoded values in this
+   * instance.
+   *
+   * @return true if the key successfully decoded, false otherwise. If false is
+   * returned, this instance is in an undefined state until the next call to
+   * `Decode()`.
+   */
+  ABSL_MUST_USE_RESULT
+  bool Decode(absl::string_view key);
+
+  /** The bundle ID for this entry. */
+  const std::string& bundle_id() const {
+    return bundle_id_;
+  }
+
+ private:
+  std::string bundle_id_;
+};
+
+/**
+ * A key in the named_queries table, storing the query name for each entry.
+ */
+class LevelDbNamedQueriesKey {
+ public:
+  /**
+   * Creates a key prefix that points just before the first key of the table.
+   */
+  static std::string KeyPrefix();
+
+  /**
+   * Creates a key that points to the key for the given query name.
+   */
+  static std::string Key(absl::string_view bundle_id);
+
+  /**
+   * Decodes the given complete key, storing the decoded values in this
+   * instance.
+   *
+   * @return true if the key successfully decoded, false otherwise. If false is
+   * returned, this instance is in an undefined state until the next call to
+   * `Decode()`.
+   */
+  ABSL_MUST_USE_RESULT
+  bool Decode(absl::string_view key);
+
+  /** The query name for this entry. */
+  const std::string& name() const {
+    return name_;
+  }
+
+ private:
+  std::string name_;
+};
+
 }  // namespace local
```

Comment:
Please add a test that verifies that the description works. Something like this:

https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/test/unit/local/leveldb_key_test.cc#L419

What you'll find, I think, is that you're missing changes to `Reader::Describe` for the new components you've added: https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/local/leveldb_key.cc#L465.

Reply:
Done.